### DISCUSSION TITLE
GCN docs

### DISCUSSION THREAD

**Discussion by meyerTUM** [pyg-team/pytorch_geometric#2177] [Q&A]
In the docs of GCN it says:

    Its node-wise formulation is given by:
    .. math::
        \mathbf{x}^{\prime}_i = \mathbf{\Theta} \sum_{j}
        \frac{1}{\sqrt{\hat{d}_j \hat{d}_i}} \mathbf{x}_j
    with :math:`\hat{d}_i = 1 + \sum_{j \in \mathcal{N}(i)} e_{j,i}`, where
    :math:`e_{j,i}` denotes the edge weight from source node :obj:`i` to target
    node :obj:`j` (default: :obj:`1`)

Shouldn't it be instead:

    \mathbf{x}^{\prime}_i = \mathbf{\Theta} \sum_{j}
    \frac{e_{j,i}}{\sqrt{\hat{d}_j \hat{d}_i}} \mathbf{x}_j
        
Only if there is an edge between node i and node j, node j should contribute, right?

**Comment by rusty1s** [comment]
You are right. That is a typo. Ofc, the `j` only iterates over the neighbors of `i`. This is now fixed in master.

**Reply by meyerTUM** [reply]
Maybe I am missing something, but shouldn't `x_j` be weighted with the edge weight `e_{j,i}` and the formula only describes the default case, where all edge weights are set to 1.0?

So, shouldn't  `e_{j,i}` be in the numerator of the fraction? Am I missing something? Thanks for taking your time! I appreciate it.

**Reply by rusty1s** [reply]
Oh yeah, you are absolutely right. I really missed that. This should now be finally fixed :) Thanks once again!
